### PR TITLE
feat: add support for headers to source transformation flows

### PR DESCRIPTION
**Pull request by koladilip**
## What are the changes introduced in this PR?

Add support for headers to source transformation flows by exposing them in event context.

## What is the related Linear task?

Resolves INT-2504

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3683/test-report.html

**Review by yashasvibajpai — APPROVED** [review]

File: src/services/source/nativeIntegration.ts [hunk 1]
```diff
@@ -38,9 +38,12 @@ export class NativeIntegrationSourceService implements SourceService {
     const respList: SourceTransformationResponse[] = await Promise.all<FixMe>(
       sourceEvents.map(async (sourceEvent) => {
         try {
+          const newSourceEvent = sourceEvent as { headers: any };
+          const { headers } = newSourceEvent;
```
Comment (utsabc) [inline src/services/source/nativeIntegration.ts:42]: Lets add a basic type for source events from server?

File: src/services/source/nativeIntegration.ts [hunk 2]
```diff
@@ -38,9 +38,12 @@ export class NativeIntegrationSourceService implements SourceService {
...
+          delete newSourceEvent.headers;
           const respEvents: RudderMessage | RudderMessage[] | SourceTransformationResponse =
-            await sourceHandler.process(sourceEvent);
-          return SourcePostTransformationService.handleSuccessEventsSource(respEvents);
+            await sourceHandler.process(newSourceEvent);
+          return SourcePostTransformationService.handleSuccessEventsSource(respEvents, { headers });
```
Comment (utsabc) [inline src/services/source/nativeIntegration.ts:46]: Do we foresee any other use case in future where headers might be needed during source transformation i.e in `process` method?

Comment (koladilip) [reply src/services/source/nativeIntegration.ts:46]: possible but less likely for `x-` headers

Comment (utsabc) [reply src/services/source/nativeIntegration.ts:46]: I see

File: src/services/source/nativeIntegration.ts (hunk 1 above)
Comment (koladilip) [reply src/services/source/nativeIntegration.ts]: Added

**Review by utsabc — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**